gateway-js/src/executeQueryPlan.ts

```diff
@@ -49,12 +52,42 @@ interface ExecutionContext {
   errors: GraphQLError[];
 }
 
+function makeIntrospectionQueryDocument(introspectionSelection: FieldNode): DocumentNode {
+  return {
+    kind: Kind.DOCUMENT,
+    definitions: [
+      {
+        kind: Kind.OPERATION_DEFINITION,
+        operation: OperationTypeNode.QUERY,
+        selectionSet: {
+          kind: Kind.SELECTION_SET,
+          selections: [ introspectionSelection ],
+        }
+      }
+    ],
+  };
+}
+
+function executeIntrospection(
+  schema: GraphQLSchema,
+  introspectionSelection: FieldNode,
+): any {
+  const { data } = executeSync({
+    schema,
+    document: makeIntrospectionQueryDocument(introspectionSelection),
+    rootValue: {},
+  });
+  assert(data, () => `Introspection query for ${JSON.stringify(introspectionSelection)} should not have failed`);
+  return data[introspectionSelection.name.value];
+}
+
 export async function executeQueryPlan(
   queryPlan: QueryPlan,
   serviceMap: ServiceMap,
   requestContext: GatewayGraphQLRequestContext,
   operationContext: OperationContext,
   supergraphSchema: GraphQLSchema,
+  apiSchema: Schema,
```

Comment:
Note to reviewers: we do add an argument to an exported method, and I'll mention it in the changelog. We already added the `supergraphSchema` argument in a previous release the same way so I don't think it's an issue.

In general, the arguments of this method are a bit messy now because we pass the API schema as a `Schema` but we also have that same schema in the `OperationContext` as a `GraphQLSchema`, and while we do want to pass both versions to avoid costly conversions, we could maybe clean things up. I'll leave that for a followup though.